app/scripts/modules/core/delivery/create/CreatePipelineModal.tsx

```diff
@@ -0,0 +1,252 @@
+    delete template.id;
+
+    if (command.strategy) {
+      template.strategy = true;
+      template.limitConcurrent = false;
+    }
+
+    this.setState({submitting: true});
+    pipelineConfigService.savePipeline(template as IPipeline)
+      .then(() => this.onSaveSuccess(template as IPipeline), this.onSaveFailure);
+  };
+
+  private onSaveSuccess = (template: IPipeline): void => {
+    const application = this.props.application;
+    template.isNew = true;
+    application.getDataSource('pipelineConfigs').refresh().then(() => {
+      const newPipeline = (template.strategy ?
+                          (application.strategyConfigs.data as IPipeline[]) :
+                           application.getDataSource('pipelineConfigs').data).find(config => config.name === template.name);
+      if (!newPipeline) {
+        $log.warn('Could not find new pipeline after save succeeded.');
+        this.setState({
+          saveError: true,
+          errorMessage: 'Sorry, there was an error retrieving your new pipeline. Please refresh the browser.',
+          submitting: false,
+        });
+      } else {
+        this.setState(this.getDefaultState());
+        this.props.pipelineSavedCallback(newPipeline.id);
+      }
+    });
+  };
+
+  private onSaveFailure = (response: IHttpPromiseCallbackArg<{message: string}>): void => {
+    $log.warn(response);
+    this.setState({
+      submitting: false,
+      saveError: true,
+      errorMessage: (response && response.data && response.data.message) || 'No message provided',
+    });
+  };
+
+  public close = (): void => {
+    this.setState(this.getDefaultState());
+    this.props.showCallback(false);
+  };
+
+  private handleTypeChange = (option: Select.Option): void => {
+    this.setState({command: Object.assign({}, this.state.command, {strategy: option.value})});
+  };
+
+  private handleNameChange = (e: React.ChangeEvent<HTMLInputElement>): void => {
+    this.setState({command: Object.assign({}, this.state.command, {name: e.target.value})});
+  };
+
+  private handleTemplateChange = (option: Select.Option): void => {
+    const template = this.state.templates.find(t => t.name === option.value);
+    this.setState({command: Object.assign({}, this.state.command, {template})});
+  };
+
+  private handleErrorDismiss = (): void => {
+    this.setState({saveError: false});
+  };
+
+  private templateOptionRenderer = (option: Select.Option) => {
+    const template = this.state.templates.find(t => t.name === option.value);
+    return (
+      <div>
+        <h5>{template.name}</h5>
+        {template.stages.length > 0 && (
+          <div className="small">
+            <b>Stages: </b>
+            <ul>
+              {template.stages.map(stage => (<li key={stage.refId}>{stage.name || stage.type}</li>))}
+            </ul>
+          </div>
+        )}
+      </div>
+    );
+  };
+
+  public render() {
+    const nameHasError: boolean = !/^[^\\\^/^?^%^#]*$/.test(this.state.command.name);
```

Comment:
with a regex this complex, it would be nice to include a comment indicating what it's testing against in a human-readable way, e.g. 
`// verify name does not include: \, ^, ?, %, #`